### PR TITLE
feat: handle unprotected session expiration

### PR DESCRIPTION
**Pull request by vardan-arm**
Internal link: https://app.asana.com/0/1200588672119490/1201382347168573

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,5 @@ yarn-error.log
 package-lock.json
 
 codeqldb
+
+coverage
```
Comment (johnny243) [inline .gitignore:54]:
```suggestion
coverage

```

File: package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "lint": "eslint --fix app/assets/javascripts",
     "tsc": "tsc --project app/assets/javascripts/tsconfig.json",
     "test": "jest --config app/assets/javascripts/jest.config.js",
+    "test:coverage": "yarn test --coverage",
```
Comment (johnny243) [inline package.json:21]: This is failing for me. Is it ready to be used?

File: app/assets/javascripts/views/editor/editor_view.ts [hunk 3]
```diff
@@ -261,14 +264,63 @@ class EditorViewCtrl extends PureViewCtrl<unknown, EditorState> {
           desc: 'Changes not saved',
         });
         break;
+      case ApplicationEvent.ProtectionSessionExpiryDateChanged: {
+        if (!this.note.protected) {
+          return;
+        }
+        if (data.protectionDuration === ProtectionSessionDurations[0].valueInSeconds) {
+          await this.setTimerForNoteProtection(DURATION_TO_POSTPONE_PROTECTED_NOTE_LOCK_WHILE_EDITING);
+          // TODO: this is required when we select the protected note for the first time and set "Don't remember" -
```
Comment (johnny243) [inline app/assets/javascripts/views/editor/editor_view.ts:273]: Are these tests going to be included in this PR?

File: app/assets/javascripts/views/editor/editor_view.ts [hunk 4]
```diff
@@ -261,14 +264,63 @@ class EditorViewCtrl extends PureViewCtrl<unknown, EditorState> {
...
+        break;
+      }
     }
   }
 
+  async showOrPostponeProtectionScreen() {
+    const protectionExpiryDate = this.application.getProtectionSessionExpiryDate();
+    const now = Date.now();
+
+    // TODO: (Vardan) write tests for unprotected notes as well
```
Comment (johnny243) [inline app/assets/javascripts/views/editor/editor_view.ts:289]: Same here

File: app/assets/javascripts/views/editor/editor_view.ts (hunk 3 above)
Comment (vardan-arm) [reply app/assets/javascripts/views/editor/editor_view.ts:280]: No, everything will be cleaned up :)

File: app/assets/javascripts/views/editor/editor_view.ts (hunk 4 above)
Comment (vardan-arm) [reply app/assets/javascripts/views/editor/editor_view.ts:293]: The same)

File: package.json (hunk 2 above)
Comment (vardan-arm) [reply package.json:21]: It works on my side. Once PR is cleaned up, we can check it again.

**Review by moughxyz — APPROVED** [review]

**Pull request merged**